ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Worker.java

```diff
@@ -561,12 +561,12 @@ protected Boolean findNextCompactionAndExecute(boolean computeStats) throws Inte
       } catch (Throwable e) {
         LOG.error("Caught exception while trying to compact " + ci +
             ".  Marking failed to avoid repeated failures", e);
-        abortCompactionAndMarkFailed(ci, compactorTxnId, e);
+        compactorTxnId = abortCompactionAndMarkFailed(ci, compactorTxnId, e);
       }
     } catch (TException | IOException t) {
       LOG.error("Caught an exception in the main loop of compactor worker " + workerName, t);
       try {
-        abortCompactionAndMarkFailed(ci, compactorTxnId, t);
+        compactorTxnId = abortCompactionAndMarkFailed(ci, compactorTxnId, t);
```

Comment:
Why not just set it to TXN_ID_NOT_SET after calling abortComapctionAndMarkFailed?

Reply:
I thought that this way it would be more obvious to future developers that the compactorTxnId needs to be unset every time it's aborted.